mypyc/ir/ops.py

```diff
@@ -226,6 +228,37 @@ def accept(self, visitor: 'OpVisitor[T]') -> T:
         return visitor.visit_assign(self)
 
 
+class AssignMulti(Op):
+    """Assign multiple values to a Register (dest = src1, src2, ...).
+
+    This is used to initialize RArray values. It's provided to avoid
+    very verbose IR for common vectorcall operations.
+
+    Note that this interacts atypically with reference counting. We
+    assume that each RArray register is initialized exactly once
+    with this op.
```

Comment:
Maybe this can be enforced somehow, for example have a class level set of destination registers, and check in `__init__` that the destination is not in this set?

Reply:
That's something we should do once we have general validation of the generated IR. However, this is probably not the best starting point for IR validation. Anyway, the C compilation will fail if we try to initialize an array twice.